### PR TITLE
[T4A7][T14-B1] Tan Min Rong Samson

### PR DESCRIPTION
**Pull request by samsontmr**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Name.java [hunk 1]
```diff
@@ -60,5 +67,40 @@ public boolean equals(Object other) {
...
+     /**
+      * Returns true of the other name is very similar to this name.
+      * Two names are considered similar if ...
+      */
+     public boolean isSimilar(Name other) {
+         if (other == null) {
+             return false;
+         }
+         
+         return isSuperSetOfIgnoreCase(other) || isSubSetOfIgnoreCase(other);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java:80]: think how you can eliminate one of the methods by using 2 objects?

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    
+    Name name;
+    
+    @Before
+    public void setUp() throws Exception {
+        name = new Name("name n");
+    }
+
+    @Test
+    public final void test_isSimilar_sameString() throws IllegalValueException {
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/person/NameTest.java:20]: you can eliminate the word `test` from the method name; 
the test name should have 3 parts `<method under test>_<which scenario/test case>_<expected return value>`

**Comment by okkhoy** [comment]
@samsontmr 
- Looking at your commits, I can see that the order of test-develop is little messed up. I hope the PR we reviewed during the tutorial and associated discussions give an idea on how to start off with TDD.
- Please take a look at the course handouts and suggested readings given in the assignment description to get a good understanding of TDD.

**Comment by okkhoy** [comment]
@samsontmr  Some comments added. Please ack & close the PR after reading comments.

**Comment by samsontmr** [comment]
@okkhoy what do you mean?

**Comment by okkhoy** [comment]
in the true sense, TDD means you develop a test first, that fails and write code to pass that test. then write a second test; repeat the cycle.
the only way we get to know that is via your commits. ideally you should have one commit for each test you write and the code in the function that passes that test.
with just 1 commit, it is hard to verify if you followed TDD. hence that comment. 
please recollect the discussion that happened around this in the tutorial.

**Pull request closed without merging**